libs/alcotest_ext/lib/Alcotest_ext.mli

```diff
@@ -1,33 +1,57 @@
 (*
    Utilities for writing test suites that are compatible with Alcotest.
 
+   This is the only module exported by this library. Other modules are
+   either hidden or included as submodules such as 'Alcotest_ext_tag'
+   which is exposed as 'Alcotest_ext.Tag'. This allows us to:
+   - use dedicated file-modules for data structures without cramming
+     everything into this single file such as Tag module providing Tag.t
+   - hide internal modules that shouldn't be accessed by users of the library
+
+   Dune exposes only this module as long as its name matches the name of the
+   library.
+
    Alcotest API:
    https://mirage.github.io/alcotest/alcotest/Alcotest/index.html
 *)
 
-type output =
+type expected_outcome =
+  | Should_succeed
+  | Should_fail of string (* explains why we expect this test to fail *)
+
+type outcome = Succeeded | Failed
+
+type captured_output =
+  | Ignored
+  | Captured_stdout of string
+  | Captured_stderr of string
+  | Captured_stdout_stderr of string * string
+  | Captured_merged of string
+
+type result = { outcome : outcome; captured_output : captured_output }
+
+type expectation = {
+  expected_outcome : expected_outcome;
+  expected_output : (captured_output, string) Result.t;
+}
+
+type status = { expectation : expectation; result : (result, string) Result.t }
+
+type status_class = PASS | FAIL | XFAIL | XPASS | MISS
```

Comment:
maybe a comment explaining each of those categories

Reply:
I had to add a bunch of types to the interface so as to allow some 3rd-party code to translate the results to something else. I did that for semgrep-pro because it exports to Junit. Exporting to Junit is not very complicated with the Junit OCaml library but it requires access to the test statuses.

I'll add comments and hints that these types are for advanced use and subject to change.